addons/hr_holidays/models/hr_leave.py

```diff
@@ -425,9 +425,19 @@ def _get_durations(self, check_leave_type=True, resource_calendar=None):
                 continue
             hours, days = (0, 0)
             if leave.employee_id:
-                if leave.employee_id.is_flexible and leave.leave_type_request_unit in ['day','half_day']:
-                    duration = leave.date_to - leave.date_from
-                    days = ceil(duration.total_seconds() / (24 * 3600))
+                if leave.employee_id.is_flexible and leave.leave_type_request_unit in ['day', 'half_day']:
+                    if leave.leave_type_request_unit == 'half_day' and leave.request_unit_half:
+                        daily_hours = calendar.get_work_hours_count(
+                            datetime.combine(leave.date_from.date(), time.min),
+                            datetime.combine(leave.date_from.date(), time.max),
+                            False
+                        )
+                        # For a half-day leave, force days = 0.5.
+                        days = 0.5
+                        hours = daily_hours / 2 if daily_hours else 0
```

Comment:
Rather than 0, it should ensure some kind of non-null value here, otherwise half-day for some flexible employees might end up taking no time at all. For flexible employees with defined hours per day, it should be half of that. For those that are fully flexible, I'd say to keep the duration of the leave (`date_to - date_from`)